doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1748,31 +1757,39 @@ Controlling the proof flow
    impossible. In most cases, :g:`P` or :g:`∼P` is one of the hypotheses of the
    local context.
 
-.. tacn:: contradiction
-   :name: contradiction
+.. tacn:: contradiction {? @one_term_with_bindings }
 
-   This tactic applies to any goal. The contradiction tactic attempts to
-   find in the current context (after all intros) a hypothesis that is
-   equivalent to an empty inductive type (e.g. :g:`False`), to the negation of
-   a singleton inductive type (e.g. :g:`True` or :g:`x=x`), or two contradictory
-   hypotheses :g:`P` and :g:`not P`.
+   Tries to prove the current goal by finding a contradiction.  If
+   :n:`@one_term_with_bindings` is not provided, the tactic first does a
+   :tacn:`intros`
+   then succeeds if there is a hypothesis :n:`False` in the updated context, or
+   if there is a pair of hypotheses where one is simple negation of the other
+   (i.e. :n:`P` and not :n:`~P`).
 
-   .. exn:: No such assumption.
-      :undocumented:
+   Use the :tacn:`discriminate` tactic to prove the current goal when there
+   is a hypothesis with an impossible structural equality such as
+   :n:`0 = 1`.
 
-.. tacv:: contradiction @ident
+   If :n:`@one_term_with_bindings` has the type :n:`False`, the tactic
+   proves the current goal.  If the type is a negation, e.g. :n:`~P`, then
+   the current goal is replaced with :n:`P`.  (However, the converse doesn't
+   work, i.e. if the type is :n:`P` and the goal is :n:`~P`.)
 
-   The proof of False is searched in the hypothesis named :n:`@ident`.
+   .. todo PR: examples?  Or maybe it's simple enough that they're not needed
+
+   .. exn:: No such assumption.
+      :undocumented:
```

Comment:
Remove this one (it is already documented elsewhere in the chapter) and add "Not a contradiction" instead.